state/executor_test.go

```diff
@@ -210,7 +210,11 @@ func TestApplyBlock(t *testing.T) {
 
 	// Apply the block with an invalid commit
 	err = executor.Validate(state, block, invalidCommit, proposer)
-	require.Error(types.ErrInvalidSignature)
+
+	// FIXME: This test didn't check for specific error. It was just checking for error.
```

Comment:
not sure I understand. you're saying that `require.Error(types.ErrInvalidSignature)` failed? if so how did the test pass? or what am I missing?

Reply:
`require.Error(types.ErrInvalidSignature)` is wrong. it checks that the type `types.ErrInvalidSignature` is of `error` type.

checking for specific error using `require.ErrorIs` fails.

I've changed it to check for any error
`require.Error(err)`

if u check
require.ErrorIs